controllers/dscinitialization/servicemesh_setup.go

```diff
@@ -36,23 +53,68 @@ func (r *DSCInitializationReconciler) configureServiceMesh(instance *dsciv1.DSCI
 func (r *DSCInitializationReconciler) removeServiceMesh(instance *dsciv1.DSCInitialization) error {
 	// on condition of Managed, do not handle Removed when set to Removed it trigger DSCI reconcile to clean up
 	if instance.Spec.ServiceMesh.ManagementState == operatorv1.Managed {
-		serviceMeshFeatures := feature.ClusterFeaturesHandler(instance, configureServiceMeshFeatures())
-
-		if err := serviceMeshFeatures.Delete(); err != nil {
-			r.Log.Error(err, "failed deleting service mesh resources")
-			r.Recorder.Eventf(instance, corev1.EventTypeWarning, "DSCInitializationReconcileError", "failed deleting service mesh resources")
+		capabilities := []*feature.HandlerWithReporter[*dsciv1.DSCInitialization]{
+			r.serviceMeshCapability(instance, serviceMeshCondition(status.RemovedReason, "Service Mesh removed")),
+		}
 
+		authzCapability, err := r.authorizationCapability(instance, authorizationCondition(status.RemovedReason, "Service Mesh Authorization removed"))
+		if err != nil {
 			return err
 		}
+
+		capabilities = append(capabilities, authzCapability)
```

Comment:
so the setup in both `configureServiceMesh`() and `removeServiceMesh`() are the same , except message  is different?